### PR TITLE
build: harvey_binary type

### PR DESCRIPTION
**Pull request by sevki**
this is going to look like a big commit but it isn't. All it does
is move binary configurations for builds to sys/src/harvey.BUILD
where it defines a harvey_binary macro.

the macro should be used for all std executables by

	load("//sys/src/harvey.BUILD", "harvey_build")

	harvey_binary(
		name="echo",
		srcs=[
			"echo.c"
		],
	)

this PR also aims to clean up the strip commands in `sys/src/9/amd64/BUILD` by moving the option into the `cc_binary` rule and stripping by default for `harvey_binary` macro

### CODE REVIEW

**Comment by elbing** [comment]
This made the kernel boot until ipconfig, isn't it? lgtm?

**Comment by sevki** [comment]
yup.

**Comment by rminnich** [comment]
do you still need the old build tool to build the rest of the tree?

**Comment by sevki** [comment]
nope, don't need it

**Comment by sevki** [comment]
@elbing if you want to LGTM it's fine

**Comment by sevki** [comment]
image

**Comment by sevki** [comment]
image

**Comment by sevki** [comment]
can someone LGTM this?

File: sys/src/cmd/BUILD [hunk 1]
```diff
@@ -1,195 +1,86 @@
...
 
-# Order of libs
-# "/$ARCH/lib/libString.a",
-# "/$ARCH/lib/lib9p.a",
-# "/$ARCH/lib/libndb.a",
-# "/$ARCH/lib/libauth.a",
-# "/$ARCH/lib/libbio.a",
-# "/$ARCH/lib/libip.a",
-# "/$ARCH/lib/libc.a"
+CMD_SRCS = [
```
Comment (rminnich) [inline sys/src/cmd/BUILD]: Why not just put them all in here?

File: sys/src/cmd/BUILD [hunk 2]
```diff
@@ -1,195 +1,86 @@
...
-    ]
+[harvey_binary(
+	name=c[:-2],
+	srcs=[c],
+) for c in CMD_SRCS]
 
-group(
-    name="cmds",
-    deps=CMDS
+harvey_binary(
```
Comment (rminnich) [inline sys/src/cmd/BUILD]: If we make the list above complete, why do we need these?

**Comment by rminnich** [comment]
LGTM
it's a start!

**Comment by sevki** [comment]
@rminnich keeping everything the same and focusing just on changing the target types

**Pull request merged**